pkg/base/util/signal_handler.go

```diff
@@ -24,15 +24,28 @@ import (
 	"github.com/dell/csi-baremetal/pkg/base/rpc"
 )
 
-//SetupSignalHandler set up channel for SIGTERM signal, when SIGTERM is caught function try to shutdown service
-func SetupSignalHandler(server *rpc.ServerRunner) {
-	sigint := make(chan os.Signal, 1)
+// SetupSIGTERMHandler tries to shutdown service, when SIGTERM is caught
+func SetupSIGTERMHandler(server *rpc.ServerRunner) {
+	setupSignalHandler(syscall.SIGTERM)
+	// We received an interrupt signal, shut down.
+	server.StopServer()
+}
 
-	signal.Notify(sigint, syscall.SIGTERM)
+// SetupSIGHUPHandler tries to make cleanup, when SIGHUP is caught
+func SetupSIGHUPHandler(cleanupFn func()) {
+	setupSignalHandler(syscall.SIGHUP)
+	// We received an SIGHUP signal, clean up.
+	if cleanupFn != nil {
+		cleanupFn()
+	}
+}
 
-	//Wait SIGTERM handler
-	<-sigint
+// setupSignalHandler sets up channel for signal
+func setupSignalHandler(sig syscall.Signal) {
+	signalChan := make(chan os.Signal, 1)
 
-	// We received an interrupt signal, shut down.
-	server.StopServer()
+	signal.Notify(signalChan, sig)
+
+	//Wait signal
+	<-signalChan
```

Comment:
Can you add debug logs about the signal we receive, will be helpful in the testing process.